api/src/main/java/javax/enterprise/event/TransactionPhase.java

```diff
@@ -26,10 +26,14 @@
  * Transactional observer methods are observer methods which receive event notifications during the before or after completion
  * phase of the transaction in which the event was fired. If no transaction is in progress when the event is fired, they are
  * notified at the same time as other observers.
+ * If the transaction is in an illegal state or already rolled back when the event is fired, the {@link #BEFORE_COMPLETION},
```

Comment:
What about _"If a transaction is in progress but a JTA `Synchronization` callback cannot be registered (e.g. the transaction is marked for rollback), the BEFORE_COMPLETION..."_